plugins/inputs/redfish/redfish.go

```diff
@@ -0,0 +1,394 @@
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode == 200 {
+		body, _ := ioutil.ReadAll(resp.Body)
+		jsonErr := json.Unmarshal(body, &h)
+		if jsonErr != nil {
+			return fmt.Errorf("error parsing input: %v", jsonErr)
+		}
+	} else {
+		return fmt.Errorf("received status code %d (%s), expected 200",
+			resp.StatusCode,
+			http.StatusText(resp.StatusCode))
+	}
+	return nil
+}
+
+func getLocation(host, Username, password, id string) error {
+	r := fmt.Sprint(host, "/redfish/v1/Chassis/", id, "/")
+	client := &http.Client{}
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	req, err := http.NewRequest("GET", r, nil)
+	req.Header.Add("Authorization", "Basic "+basicAuth(Username, password))
+	req.Header.Set("Accept", "application/json")
+	req.Header.Set("Content-Type", "application/json")
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode == 200 {
+		body, _ := ioutil.ReadAll(resp.Body)
+		jsonErr := json.Unmarshal(body, &l)
+		if jsonErr != nil {
+			return fmt.Errorf("error parsing input: %v", jsonErr)
+		}
+	} else {
+		return fmt.Errorf("received status code %d (%s), expected 200",
+			resp.StatusCode,
+			http.StatusText(resp.StatusCode))
+	}
+
+	return nil
+}
+
+func (r *Redfish) Description() string {
+	return "Read CPU, Fans, Powersupply and Voltage metrics of Dell/HP hardware server through redfish APIs"
+}
+
+var redfishConfig = `
+## Server OOB-IP
+host = "https://192.0.0.1"
+
+## Username,Password for hardware server
+basicauthusername = "test"
+basicauthpassword = "test"
+## Server Vendor(dell or hp)
+server= "dell"
+## Resource Id for redfish APIs
+id="System.Embedded.1"
+
+## Amount of time allowed to complete the HTTP request
+# timeout = "5s"
+`
+
+func (r *Redfish) SampleConfig() string {
+	return redfishConfig
+}
+
+func (r *Redfish) SetParser(parser parsers.Parser) {
+	r.parser = parser
+}
+
+func (r *Redfish) Init() error {
+	return nil
+}
+
+func (r *Redfish) Gather(acc telegraf.Accumulator) error {
+
+	if len(r.Host) > 0 && len(r.BasicAuthUsername) > 0 && len(r.BasicAuthPassword) > 0 && len(r.Server) > 0 && len(r.Id) > 0 && (r.Server == "dell" || r.Server == "hp") {
+		err := getThermal(r.Host, r.BasicAuthUsername, r.BasicAuthPassword, r.Id)
+		if err != nil {
+			return err
+		}
+		err = getHostname(r.Host, r.BasicAuthUsername, r.BasicAuthPassword, r.Id)
+		if err != nil {
+			return err
+		}
+
+		err = getPower(r.Host, r.BasicAuthUsername, r.BasicAuthPassword, r.Id, r.Server)
+		if err != nil {
+			return err
+		}
+
+		if r.Server == "dell" {
+			err = getLocation(r.Host, r.BasicAuthUsername, r.BasicAuthPassword, r.Id)
+			if err != nil {
+				return err
+			}
+
+		}
+
+		for i := 0; i < len(t.Temperatures); i++ {
+			//  Tags
+			//			tags := map[string]string{"Name": t.Temperatures[i].Name}
+			tags := map[string]string{"OOBIP": r.Host, "Name": t.Temperatures[i].Name, "Hostname": h.Hostname}
```

Comment:
Rather than building tags and fields manually, you might have an easier time with `metric.New()` and `AddField()`, which won't require type conversions.

Reply:
do you mean usage of strconv in the code for type conversion can be replaced with metric.New() and AddField()?

Reply:
Yeah, but I'll retract that. This is fine as you have it.